### DISCUSSION TITLE
Will the bank send addenda files for all entry details in ACH prenotifications?

### DISCUSSION THREAD

**Discussion by samyje** [moov-io/ach#998] [Q&A] [answered]
We have to send 1000s of entry detail records for prenotifications to the bank, will bank send addenda record for all the entry detail records we are sending or just the ones which has return codes. Also, what is the position for return code the bank send us in the addenda record? What is the maximum entry detail records we can send in a batch? What is the maximum number of batches we can send in a file? 

Your contributions and answers are really helpful. Thank you!

**Comment by adamdecaf** [comment, marked as answer]
Hey @samyje! Thanks for the questions. I've answered them below. 

> We have to send 1000s of entry detail records for prenotifications to the bank, will bank send addenda record for all the entry detail records we are sending or just the ones which has return codes. 

I don't think your bank would send the [05 addenda](https://github.com/moov-io/ach/blob/master/addenda05.go) record if you specify it, but that's a question for your bank. Your bank will include [99 addenda records](https://github.com/moov-io/ach/blob/master/addenda99.go) for returns.

Both 05 and 99 addenda records can be [parsed and consumed](https://github.com/moov-io/ach/blob/master/entryDetail.go#L81) using this library. (hint: `ach.ReadFile(path)`)

> Also, what is the position for return code the bank send us in the addenda record? 
You can grab the [return code from the 99 addenda record](https://github.com/moov-io/ach/blob/v1.13.0/addenda99.go#L107) using this logic / code. 

> What is the maximum entry detail records we can send in a batch? What is the maximum number of batches we can send in a file?
Nacha official guidelines limit files to 10,000 lines, so you can combine batches/entries anyway you want under that limit. Keep in mind that a Batch header and control are two lines. 

Our library doesn't quite support splitting one big `struct` into multiple files, but it does support merging lots of `*ach.File` objects into a few files as possible. We use [`ach.MergeFiles` a lot in production](https://github.com/moov-io/ach/blob/v1.13.0/merge.go#L27-L38)

**Reply by samyje** [reply]
Thank you so much for the answers.
